File: 0001~0500/0191. Number of 1 Bits/main.py
"""
Source: https://leetcode.com/problems/number-of-1-bits/description/
Date: 2023/1/25
Skill:
Runtime: 34 ms, faster than 71.90%
Memory Usage: 13.9 MB, less than 45.57%
Time complexity:
Space complexity:
Constraints:
    The input must be a binary string of length 32.
"""

import math
from typing import List
from collections import defaultdict, deque
from heapq import heapify, heappush, heappop, nsmallest
import heapq
import functools


class Solution:
    def hammingWeight(self, n: int) -> int:
        return sum(n & 1<<i != 0 for i in range(32))

File: 0001~0500/0191. Number of 1 Bits/main.cpp
/**
 * Source: t.ly/340YB
 * Date: 2023/11/29
 * Skill:
 * Ref:
 * Runtime: 0 ms, faster than 100.00% of C++ online submissions
 * Memory Usage: 6.29 MB, less than 94.23% of C++ online submissions
 * Time complexity:
 * Space complexity:
 * Constraints:
 *
 */


#include <iostream>
#include <vector>
#include <algorithm>
#include <map>
#include <set>
#include <unordered_set>
#include <queue>
#include <numeric>
#include <tuple>
#include <stack>
#include <bitset>
#include <functional>
#include <list>


using namespace std;


class Solution {
public:
    int hammingWeight(uint32_t n) {
        int res = 0;
        while (n) {
            res ++;
            n &= n - 1;
        }
        return res;
    }
};